### PR TITLE
Esp json

### PR DESCRIPTION
**Pull request by NathanThus**

### CODE REVIEW

**Comment by NathanThus** [comment]
@TomVer99 My tests for JsonParsing aren't being added to the list of actions, despite the workflow.yml file having the correct directoy, do you have any ideas?

**Review by TomVer99 — CHANGES_REQUESTED** [review]
No review done yet, this is to allow me to request a rereview from myself to trigger the tests.

File: Source/Master/lib/JsonBuilder/JsonBuilder.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#include "JsonBuilder.hpp"
+#include "json.hpp"
+using json = nlohmann::json;
+
+JsonBuilder::JsonBuilder()
```
Comment (TomVer99) [inline Source/Master/lib/JsonBuilder/JsonBuilder.cpp:5]: Constructor without contents? Use `= default` in header?
"A defaulted default constructor is specifically defined as being the same as a user-defined default constructor with no initialization list and an empty compound statement."
https://stackoverflow.com/questions/20828907/the-new-syntax-default-in-c11

File: Source/Master/workflow.yml
Comment (TomVer99) [inline Source/Master/workflow.yml:1]: Action should be specified in `FindIt/.github/workflows/CI.yml` not here.

**Review by TomVer99 — CHANGES_REQUESTED** [review]
If there is time, check what parameters can be converted to const references.

File: Source/Master/lib/JsonBuilder/JsonBuilder.cpp (hunk 1 above)
Comment (NathanThus) [reply Source/Master/lib/JsonBuilder/JsonBuilder.cpp]: Fixed

**Comment by NathanThus** [comment]
Already fixed the styling error, happened due to tests changing.
Waiting to commit, need some assistance with integrating the unit tests.

**Review by NathanThus — COMMENTED** [review]
:)

**Comment by NathanThus** [comment]
@TomVer99 Please test :)

**Review by TomVer99 — CHANGES_REQUESTED** [review]
Some types are changed, is this supposed to be?
For example in JsonBuilder.hpp @ line 17 & 25.
If this is supposed to be then its fine.

Rerequest a review to run the workflows.

**Comment by NathanThus** [comment]
@TomVer99 Only two tests are detected :)

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -31,29 +31,26 @@ jobs:
...
-
-  #     - name: Install PlatformIO Core
-  #       run: pip install --upgrade platformio
-
-  #     - name: Test
-  #       working-directory: ./Source/Node/
-  #       run: platformio test --environment native -f unit -v
+
+  Node:
+    runs on: ubuntu-latest
```
Comment (TomVer99) [inline .github/workflows/CI.yml:36]: Spelling error, added review to allow for rereview.

**Review by TomVer99 — CHANGES_REQUESTED** [review]

**Review by TomVer99 — APPROVED** [review]
looks fine.

**Pull request merged**